manifests/security.pp

```diff
@@ -17,17 +17,91 @@
 # Jenkins security configuration
 #
 class jenkins::security (
-  $security_model = undef,
+  $jenkins_management_email         = '',
+  $jenkins_management_login         = '',
+  $jenkins_management_name          = '',
+  $jenkins_management_password      = '',
+  $jenkins_s2m_acl                  = false,
+  $jenkins_ssh_private_key_contents = '',
+  $jenkins_ssh_public_key_contents  = '',
+  $ldap_access_group                = '',
```

Comment:
This is a huge amount of parameters for the jenkins::security class. I suggest changing this to take a hash called `ldap_config`. You could even set the defaults in a container class like `jenkins::ldap::config` or something like that.